tests/bin/demo-addition.test.ts

```diff
@@ -0,0 +1,55 @@
+import main from '@/bin/demo-addition';
+
+const maybeMockRestore = (a: any): void =>
+  a.mockRestore && typeof a.mockRestore === 'function'
+    ? a.mockRestore()
+    : undefined;
+
+type FunctionPropertyNames<T> = {
+  [K in keyof T]: T[K] extends (...args: any[]) => any ? K : never;
+}[keyof T];
+type ArgsType<T> = T extends (...args: infer A) => any ? A : never;
+
+function spyOnImplementing<
+  T extends typeof console,
+  M extends FunctionPropertyNames<T>,
+  F extends T[M],
```

Comment:
This is all unnecessary, you should not need to reimplement these functions at all.

Tests should be simple, all you do is import utilities from the outside and run a test.

These functions here are going to add maintenance overhead and won't be useful in the future.